src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +488,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf();
```

Comment:
I  checked, it does hit this code path. Let me know if any case is missing in that test anyways.